packages/fuelet_secure_layer/lib/src/features/password/password_manager.dart

```diff
@@ -67,46 +71,184 @@ class PasswordManager {
   final IAccountsPrivateDataRepository _privateDataRepository;
   final FlutterSecureStorage _secureStorage;
   final SessionStoragePasswordManager _sessionStoragePasswordManager;
+  final TPMService _tpmService;
 
   PasswordManager(
     this._privateDataRepository,
     this._secureStorage,
     this._sessionStoragePasswordManager,
+    this._tpmService,
   );
 
   Future<bool> get passwordExists async {
-    final value = await _secureStorage.read(key: _passwordKey);
-
+    final value = await _secureStorage.read(
+      key: _passwordKey,
+      aOptions: const AndroidOptions(
+        encryptedSharedPreferences: false,
+      ),
+    );
     return value != null;
   }
 
   Future<bool> get legacyPasscodeExists async {
-    final value = await _secureStorage.read(key: _legacyPasscodeKey);
-
+    final value = await _secureStorage.read(
+      key: _legacyPasscodeKey,
+      aOptions: const AndroidOptions(
+        encryptedSharedPreferences: false,
+      ),
+    );
     return value != null;
   }
 
   Future<void> setPassword(String password) async {
     validatePassword(password);
     final secretToStore = await _createPasswordSecret(password);
-    await _secureStorage.write(key: _passwordKey, value: secretToStore);
-    await _secureStorage.delete(key: _legacyPasscodeKey);
+    await _secureStorage.write(
+      key: _passwordKey,
+      value: secretToStore,
+      aOptions: const AndroidOptions(
+        encryptedSharedPreferences: false,
+      ),
+    );
+    await _secureStorage.delete(
+      key: _legacyPasscodeKey,
+      aOptions: const AndroidOptions(
+        encryptedSharedPreferences: false,
+      ),
+    );
+    await _sessionStoragePasswordManager.storeSessionStoragePassword(password);
+  }
 
+  Future<void> setSessionStoragePassword(String password) async {
     await _sessionStoragePasswordManager.storeSessionStoragePassword(password);
   }
 
-  Future<AuthorizationResponse> authorize(String input) async {
-    final passwordString = await _secureStorage.read(key: _passwordKey);
-    // If passwordString is null, it means that the user might be using the legacy passcode policy
-    if (passwordString == null) {
-      final passcode = await _secureStorage.read(key: _legacyPasscodeKey);
-      if (passcode == null) {
-        return AuthorizationResponse.noPassword;
+  Future<void> storePasswordForBiometry(String password) async {
+    validatePassword(password);
+
+    if (Platform.isIOS) {
+      final enclave = SecureEnclave();
+
+      final keyResult = await _tpmService.generateKeyPair(_secureEnclaveKey);
```

Comment:
TmpService for ios is secure enclave, so probably we don't need here both secure enclave and tpm service. Let's leave one of them only. If we go for TpmService, then it should be implemented for android as well, and probably we will not need different branches for each platform here